packages/common/src/api/tan-query/jupiter/useSwapTokens.ts

```diff
@@ -0,0 +1,375 @@
+        const {
+          tokenLedgerInstruction,
+          swapInstruction,
+          addressLookupTableAddresses
+        } = await jupiterInstance.swapInstructionsPost({
+          swapRequest: {
+            quoteResponse: quoteResult.quote,
+            userPublicKey: userPublicKey.toBase58(),
+            destinationTokenAccount: preferredJupiterDestination,
+            wrapAndUnwrapSol: wrapUnwrapSol,
+            useSharedAccounts: true
+          }
+        })
+
+        const jupiterInstructions = convertJupiterInstructions([
+          tokenLedgerInstruction,
+          swapInstruction
+        ])
+
+        updateJupiterAtaCreationFeePayer(
+          jupiterInstructions,
+          userPublicKey,
+          feePayer
+        )
+        instructions.push(...jupiterInstructions)
+
+        // --- 3d. Identify Actual Jupiter Destination & Potential Jupiter-Created Temporary ATA ---
+        const actualJupiterDestination =
+          findActualJupiterDestination(jupiterInstructions)
+        const jupiterTemporarySetupAta = outputTokenConfig
+          ? findJupiterTemporarySetupAta(
+              jupiterInstructions,
+              outputTokenConfig.mintAddress,
+              actualJupiterDestination
+            )
+          : undefined
+
+        // --- 3e. Handle Output Token (if user bank managed and not directly deposited by Jupiter) ---
+        let wasOutputDepositedToUserBank = false
+        if (
+          outputUserBankAddress &&
+          actualJupiterDestination?.equals(outputUserBankAddress)
+        ) {
+          wasOutputDepositedToUserBank = true
+        }
+
+        const needsManualTransferToOutputUserBank =
+          isOutputUserBankManaged &&
+          actualJupiterDestination &&
+          !wasOutputDepositedToUserBank
+
+        if (needsManualTransferToOutputUserBank) {
+          try {
+            await addAtaToUserBankInstructions({
+              tokenInfo: outputTokenConfig!,
+              userPublicKey,
+              ethAddress: ethAddress!,
+              amountLamports: BigInt(quoteResult.outputAmount.amount),
+              sourceAta: actualJupiterDestination!,
+              sdk,
+              feePayer,
+              instructions
+            })
+            wasOutputDepositedToUserBank = true
+          } catch (error: any) {
+            reportToSentry({
+              name: 'JupiterSwapOutputPostProcessError',
+              error,
+              feature: Feature.TanQuery,
+              additionalInfo: { params }
+            })
+          }
+        }
+
+        // --- 3f. Add Cleanup Instructions for Temporary ATAs ---
+        const atasToClose: PublicKey[] = []
+
+        // Add source ATA if it's temporary
+        if (sourceAtaForJupiter && isSourceAtaTemporary) {
+          atasToClose.push(sourceAtaForJupiter)
+        }
+
+        // Determine if Jupiter setup ATA should be closed
+        const shouldCloseJupiterSetupAta =
+          jupiterTemporarySetupAta &&
```

Comment:
we shouldn't have any setup ATAs from jupiter, since we're ignoring that return value from the API, right?